src/globus_sdk/services/auth/client/base.py

```diff
@@ -203,6 +197,106 @@ def _convert_listarg(
             self.get("/v2/api/identities", query_params=query_params)
         )
 
+    def get_identity_providers(
+        self,
+        *,
+        domains: t.Iterable[str] | str | None = None,
+        ids: t.Iterable[UUIDLike] | UUIDLike | None = None,
+        query_params: dict[str, t.Any] | None = None,
+    ) -> GetIdentityProvidersResponse:
+        r"""
+        Given ``domains=...`` or (exclusive) ``ids=...``,
+        look up information for the matching set of identity providers.
```

Comment:
I think this first line could be clearer. That `domains` and `ids` are mutually exclusive is covered by their parameter documentation.

```suggestion
        Look up information about identity providers by domains or by IDs.
```

Reply:
Yep, I like this a lot better. (Aside: let's revisit the `get_identities` docstring, which is similarly stated.)